firewall.tf

```diff
@@ -0,0 +1,37 @@
+locals {
+  firewall_rules = module.this.enabled && var.firewall_rules != null ? {
+    for indx, rule in flatten(var.firewall_rules) :
+    format("%s-%s-%s",
+      rule.action,
+      md5(rule.expression),
+      lookup(rule, "description", null) == null ? md5(format("Managed by Terraform #%d", indx)) : md5(rule.description),
```

Comment:
This is too much for a key. Make `firewall_rules` a `map(any)` to begin with.

Reply:
> Make firewall_rules a map(any) to begin with.
What you want to use as key? `action` isn't uniq, `expression` and `description` can contains spaces and special characters... each think how to call some block sounds not cool. IMHO just logically list of maps is better than map of maps for declaration. Here's like with rest-API the key of object should be generated by backend.